File: easy/matrix_rotation.pl
use strict;
use integer;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    chomp $line;
    my @s = split( / /, $line );
    my @r;
    my $n = sqrt( scalar @s );
    for ( my $i = 0 ; $i < $n ; $i++ ) {
        for ( my $j = $n - 1 ; $j >= 0 ; $j-- ) {
            push @r, $s[ $j * $n + $i ];
        }
    }
    printf( "%s\n", join( ' ', @r ) );
}
close(INFILE);

File: easy/matrix_rotation.rb
File.open(ARGV[0]).each_line do |line|
  s = line.split
  t = []
  l = Math.sqrt(s.length).to_i
  0.upto(l - 1) do |x|
    (l - 1).downto(0) do |y|
      t << s[y * l + x]
    end
  end
  puts t.join(' ')
end
